src/gpl/src/nesterovBase.cpp

```diff
@@ -892,102 +898,65 @@ NesterovBaseCommon::NesterovBaseCommon(NesterovBaseVars nbVars,
   pbc_ = std::move(pbc);
   log_ = log;
 
-  // gCellStor init
-  gCellStor_.reserve(pbc_->placeInsts().size());
-
-  for (auto& inst : pbc_->placeInsts()) {
-    gCellStor_.emplace_back(inst);
+  uint gcell_id = 0;
+  uint gpin_id = 0;
+  uint gnet_id = 0;
+  for (auto& gpl_inst : pbc_->placeInsts()) {
+    std::shared_ptr<GCell> gCell
+        = std::make_shared<GCell>(gpl_inst, gcell_id++);
+    gCellMap_[gCell->instance()] = gCell;
+    newGCells_.insert(std::move(gCell));
   }
 
-  // TODO:
-  // at this moment, GNet and GPin is equal to
-  // Net and Pin
-
-  // gPinStor init
-  gPinStor_.reserve(pbc_->pins().size());
-  for (auto& pin : pbc_->pins()) {
-    GPin myGPin(pin);
-    gPinStor_.push_back(myGPin);
+  for (auto& gpl_pin : pbc_->pins()) {
+    std::shared_ptr<GPin> gPin = std::make_shared<GPin>(gpl_pin, gpin_id++);
+    gPinMap_[gPin->pin()] = gPin;
+    gPins_.insert(std::move(gPin));
   }
 
-  // gNetStor init
-  gNetStor_.reserve(pbc_->nets().size());
-  for (auto& net : pbc_->nets()) {
-    GNet myGNet(net);
-    gNetStor_.push_back(myGNet);
+  for (auto& gpl_net : pbc_->nets()) {
+    std::shared_ptr<GNet> gNet = std::make_shared<GNet>(gpl_net, gnet_id++);
+    gNetMap_[gNet->net()] = gNet;
+    gNets_.insert(std::move(gNet));
   }
 
-  // gCell ptr init
-  gCells_.reserve(gCellStor_.size());
-  for (auto& gCell : gCellStor_) {
-    if (!gCell.isInstance()) {
+  //#pragma omp parallel for num_threads(num_threads_)
+  for (auto& gCell : newGCells_) {
+    if (gCell->isFiller()) {
       continue;
     }
-    gCells_.push_back(&gCell);
-    gCellMap_[gCell.instance()] = &gCell;
-  }
-
-  // gPin ptr init
-  gPins_.reserve(gPinStor_.size());
-  for (auto& gPin : gPinStor_) {
-    gPins_.push_back(&gPin);
-    gPinMap_[gPin.pin()] = &gPin;
-  }
-
-  // gNet ptr init
-  gNets_.reserve(gNetStor_.size());
-  for (auto& gNet : gNetStor_) {
-    gNets_.push_back(&gNet);
-    gNetMap_[gNet.net()] = &gNet;
-  }
-
-  // gCellStor_'s pins_ fill
-#pragma omp parallel for num_threads(num_threads_)
-  for (auto it = gCellStor_.begin(); it < gCellStor_.end(); ++it) {
-    auto& gCell = *it;  // old-style loop for old OpenMP
-
-    if (gCell.isFiller()) {
-      continue;
-    }
-
-    for (auto& pin : gCell.instance()->pins()) {
-      gCell.addGPin(pbToNb(pin));
+    for (auto& pin : gCell->instance()->pins()) {
+      gCell->addGPin(pbToNb(pin).get());
     }
   }
 
-  // gPinStor_' GNet and GCell fill
-#pragma omp parallel for num_threads(num_threads_)
-  for (auto it = gPinStor_.begin(); it < gPinStor_.end(); ++it) {
-    auto& gPin = *it;  // old-style loop for old OpenMP
-
-    gPin.setGCell(pbToNb(gPin.pin()->instance()));
-    gPin.setGNet(pbToNb(gPin.pin()->net()));
+  //#pragma omp parallel for num_threads(num_threads_)
+  for (auto& gPin : gPins_) {
+    gPin->setGCell(pbToNb(gPin->pin()->instance()).get());
+    gPin->setGNet(pbToNb(gPin->pin()->net()).get());
```

Comment:
.get() here should not be used (these are shared_ptrs and should be passed as such)